File: Week2/Python/RecursionProblem/Print1toN.py
"""
Print numbers from 1 to n without the help of loops. You only need to complete the function printNos() that takes N as 
parameter and prints number from 1 to N recursively.

https://www.geeksforgeeks.org/problems/print-1-to-n-without-using-loops-1587115620/1
"""

def printNos(N):
    """ T(c) -> O(n), S(c) -> O(n) """
    # base case
    if N < 1:
        return
    
    printNos(N-1)
    
    print(N, end=" ")


N = 5
printNos(N)

File: Week2/Java/RecursionProblem/Print1toN.java
public class Print1toN {

/*
 * Print numbers from 1 to N without the help of loops.
 * https://www.geeksforgeeks.org/problems/print-1-to-n-without-using-loops-1587115620/1
 * T(c) - > O(N), S(c) -> O(N)
 */
    static void print1oN(int N)
    {
        // Base case
        if(N == 0)
        {
            return;
        }
        
        // Recursion function
        print1oN(N-1);
        
        System.out.print(N+" ");
    }

    public static void main(String[] args) {
        int N = 10;
        print1oN(N);
    }
    
}
